packages/flutter/test/material/expansion_panel_test.dart

```diff
@@ -121,6 +153,25 @@ void main() {
     expect(box.size.height - oldHeight, greaterThanOrEqualTo(100.0)); // 100 + some margin
   });
 
+  testWidgets('ExpansionPanelList does not merge header when canTapOnHeader is false', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      MaterialApp(
+        home: ExpansionPanelListSemanticsTest(),
+      ),
+    );
+    await tester.pump(const Duration(milliseconds: 200));
+    SemanticsHandle handle = tester.ensureSemantics();
```

Comment:
You should probably turn on semantics before pumping the widget. That way, you probably don't need the awkward flushSemantics below.